datastore/client.go

```diff
@@ -73,6 +73,17 @@ func (dc *datastoreClient) RunQuery(ctx context.Context, in *pb.RunQueryRequest,
 	return res, err
 }
 
+func (dc *datastoreClient) RunAggregationQuery(ctx context.Context, in *pb.RunAggregationQueryRequest, opts ...grpc.CallOption) (res *pb.RunAggregationQueryResponse, err error) {
```

Comment:
We don't use raw protobuf types in the public API surface.

Reply:
Not sure if I follow. It seems the `RunQuery` above also uses raw proto type?

Reply:
The types exposed in that file aren't used directly by the client library. (I believe the file that you're adding code to is basically unused, but hadn't been able to prove that conclusively.)

Reply:
Wouldn't the file I'm adding code to get called from here [1] since it overrides the `RunAggregationQuery` call and similarly for `RunQuery` and others? 

[1] https://github.com/googleapis/google-cloud-go/blob/feb6b0a8b771875a55c75fa959e19f6f12079030/datastore/query.go#L663

Reply:
Nope, check out the [import](https://github.com/googleapis/google-cloud-go/blob/main/datastore/query.go#L30) and the [type `Client.client`](https://github.com/googleapis/google-cloud-go/blob/main/datastore/datastore.go#L59).

Reply:
Additional: Golang doesn't have the concept of overrides or shadowing in the same way that other languages do. If you have two methods which collide, the compiler simply throws an error.

Reply:
Got it, thanks for explaining.